params/superchain_test.go

```diff
@@ -59,6 +59,51 @@ func TestProtocolVersion_Compare(t *testing.T) {
 			B:   HumanProtocolVersion{0, 1, 3, 3, 3, [8]byte{3}},
 			Cmp: EmptyVersion,
 		},
+		{
+			A:   HumanProtocolVersion{0, 4, 0, 0, 0, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 1, [8]byte{}},
+			Cmp: AheadMajor,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 1, 0, 0, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 1, 0, 1, [8]byte{}},
+			Cmp: AheadMinor,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 0, 1, 0, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 1, 1, [8]byte{}},
+			Cmp: AheadPatch,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 0, 0, 2, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 1, [8]byte{}},
+			Cmp: AheadPrerelease,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 1, 0, 1, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 0, [8]byte{}},
+			Cmp: AheadPrerelease,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 0, 1, 1, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 0, [8]byte{}},
+			Cmp: AheadPrerelease,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 1, 1, 1, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 0, [8]byte{}},
+			Cmp: AheadMinor,
+		},
+		{
+			A:   HumanProtocolVersion{0, 4, 0, 2, 1, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 0, [8]byte{}},
+			Cmp: AheadPatch,
+		},
+		{
+			A:   HumanProtocolVersion{0, 5, 0, 1, 1, [8]byte{}},
+			B:   HumanProtocolVersion{0, 4, 0, 0, 0, [8]byte{}},
+			Cmp: AheadMajor,
+		},
 	}
 	for i, tc := range testCases {
 		t.Run(fmt.Sprintf("case_%d", i), func(t *testing.T) {
```

Comment:
Not strictly required since the tests aren't parallel, but for good measure:
```suggestion
	for i, tc := range testCases {
	    tc := tc
		t.Run(fmt.Sprintf("case_%d", i), func(t *testing.T) {
```